controllers/cloud/api/v1/cloudclient_types.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// CloudClientSpec defines the desired state of CloudClient
+type CloudClientSpec struct {
```

Comment:
CR is a resource not a Client

Reply:
The purpose of the CloudClient Custom Resource (CR) is to serve as the startup resource for the entire module controller. All reconcile functions need to listen to the CloudClient resource to initiate the interaction process with the cloud. The name CloudClient is indeed quite apt...